addons/hr_holidays/models/hr_leave.py

```diff
@@ -1395,7 +1414,31 @@ def _force_cancel(self, reason, msg_subtype='mail.mt_comment'):
                 body=_('The time off has been canceled: %s', reason),
                 subtype_xmlid=msg_subtype
             )
-
+            responsibles = self.env['res.partner']
+            # manager
+            if (leave.holiday_status_id.leave_validation_type == 'manager' and leave.state == 'validate') or (leave.holiday_status_id.leave_validation_type == 'both' and leave.state == 'validate1'):
+                responsibles = leave.employee_id.leave_manager_id.partner_id
+            # officer
+            elif leave.holiday_status_id.leave_validation_type == 'hr' and leave.state == 'validate':
+                responsibles = leave.holiday_status_id.responsible_id.partner_id
+            # both
+            elif leave.holiday_status_id.leave_validation_type == 'both' and leave.state == 'validate':
+                responsibles = leave.employee_id.leave_manager_id.partner_id
+                responsibles |= leave.holiday_status_id.responsible_id.partner_id
+
+            if responsibles:
```

Comment:
You could re-use the `_notify_manager` method with the subject / body as arguments